bundles/org.pitest.pitclipse.core/src/org/pitest/pitclipse/core/PitCoreActivator.java

```diff
@@ -88,6 +106,24 @@
 
     private File historyFile;
 
+
+    /**
+     * <b>Must</b> run in UI thread.
+     */
+    private void initIcons() {
+        imageRegistry = new ImageRegistry();
+        Bundle bundle = FrameworkUtil.getBundle(getClass());
+        URL url = FileLocator.find(bundle, new Path("icons/pit.gif"), null);
+        imageRegistry.put(PIT_ICON, ImageDescriptor.createFromURL(url).createImage());
+    }
+
+    /**
+     * @return Returns the pit icon.
+     */
+    public Image getPitIcon() {
+        return imageRegistry.get(PIT_ICON);
+    }
+
```

Comment:
All of this stuff should be the `.ui` plugin (the `icons/` folder + the code to initialize the image registry). You shouldn't even have been able to write this because the `.core` plugin shouldn't have a dependency toward the `org.eclipse.ui` bundle but we'll fix that in another PR.

Reply:
What does that mean for me now?

Reply:
That you should move (in this PR) the icon-related stuff to the `.ui` plugin. We can fix the extra dependency to `org.eclipse.ui` in another PR because you're not the one who introduced it. Sorry if I wasn't clear enough.

Reply:
Where should this be in the ui plugin?

Reply:
You can create a `PitclipseUiActivator` class (similar to the existing `PitLaunchUiActivator` class) and put everything in it.

Reply:
But don't forget to register the activator in `MANIFEST.MF`, e.g.:

https://github.com/pitest/pitclipse/blob/477b13eefd9401e06d1d94103b92948e24e8a58b/bundles/org.pitest.pitclipse.core/META-INF/MANIFEST.MF#L6

Reply:
This is now implemented. Please consider a merge, if @LorenzoBettini is fine, even with the strange KDE behavior.

Reply:
> All of this stuff should be the `.ui` plugin (the `icons/` folder + the code to initialize the image registry). You shouldn't even have been able to write this because the `.core` plugin shouldn't have a dependency toward the `org.eclipse.ui` bundle but we'll fix that in another PR.

Please someone open an issue about that